R/shade_p_value.R

```diff
@@ -101,14 +110,21 @@ check_shade_p_value_args <- function(obs_stat, direction, color, fill) {
   TRUE
 }
 
-geom_tail_area <- function(tail_data, fill) {
-  list(
-    ggplot2::geom_area(
-      data = tail_data, mapping = aes(x = x, y = y, group = dir),
-      fill = fill, alpha = 0.6,
-      show.legend = FALSE, inherit.aes = FALSE
-    )
+geom_tail_area <- function(tail_data, fill, ...) {
+  area_args <- c_dedupl(
+    list(
+      data = tail_data,
+      mapping = aes(x = x, y = y, group = dir),
+      fill = fill,
+      show.legend = FALSE,
+      inherit.aes = FALSE
+    ),
+    list(...),
```

Comment:
This is something that I've never understood. I thought that when you added and argument called `...` that it defaulted to being of class `list`. Is that not true?

Reply:
Well, `...` is a special language construct. In this particular piece of code `list()` *can* be omitted, as output will still be list (which is a primary goal here) because there are other list-inputs to `c_dedupl()`.
However, for a dummy wrapper `foo <- function(...) {c(...)}`, output of `foo(a = 1, b = 2)` is a numeric vector.

In this case I preferred to add it for more symmetry.

Reply:
Ah, ok, I see.